docs/indexing_samples.md

```diff
@@ -0,0 +1,51 @@
+# Adding sample indexing to the index runner
+
+## Reusable code
+
+* The event loop in `main.py`
+* The logging code in `main.py`
+* `config.py` (currently some code in `main.py` could probably be moved into `config.py` for
+  separation of concerns reasons)
+* The Kafka code in `main.py` with configurable topics
+
+## Non-reusable code
+
+* `_handle_message` in `main.py` and the methods it calls
+
+## Refactoring to avoid (nearly) duplicate code for sample indexing
+
+* Refactor the event loop so that the message handling function can be passed into the event loop
+* Separate the workspace specific code into its own module with a `main()` method that
```

Comment:
How would you run the two event loops separately? Python processes?

Reply:
That's what we're leaning towards - have the compose file run different commands. Another option would be a config option.

Reply:
You mean the docker run command on Rancher would have flags to execute different parts of the stack? That make sense.

But if you want to run the indexer on both workspace and sample events at the same time, would you start them in separate processes? Or are you thinking they would be exclusive only?

Reply:
What we've been talking about is having a single process per 'type' of event, where type is WS or Sample

Reply:
And each container would have only one process

Reply:
Nothing is set in stone though

Reply:
Separating those consumers sounds good to me. That way a really busy topic won't completely clog the queue for different events. I wonder if there will be a way to easily see what containers on Rancher have what config without having to click through a bunch of layers.